src/Server/Remote/NativeSsh.php

```diff
@@ -230,12 +206,55 @@ protected function getConnectionHash()
     protected function getMultiplexingSshOptions()
     {
         return [
-            '-o ControlMaster=auto',
-            '-o ControlPersist=5',
-            '-o ControlPath=\'' . $this->getConnectionHash() . '\'',
+            '-o' => [
+                'ControlMaster=auto',
+                'ControlPersist=5',
+                'ControlPath=' . $this->getConnectionHash(),
+            ],
         ];
     }
 
+    /**
+     * Returns final ssh connection command with configs, username and host.
+     *
+     * @return array
+     */
+    private function getSshConnectionOptions()
```

Comment:
@ossinkine In fact we need to create separate AbstractOptionsBuilder and SshOptionsBuilder class for working with all that stuff. But the PR is already too big so I'm going to do it next time.